### PR TITLE
Replace ts-node with tsx

### PR DESCRIPTION
**Pull request by AaronMoat**
~Completely untested but opening up to early feedback on the principle.~

Somewhat tested now, are we happy with major+upgrade at own risk laziness, or are there things we can do to get more confidence?

### CODE REVIEW

File: src/cli/configure/dependencies/skubaDeps.ts [hunk 1]
```diff
@@ -5,21 +5,22 @@ const DEV_DEPENDENCIES = [
   '@seek/seek-module-toolkit',
   'eslint-config-seek',
   'nodemon',
-  'tslint',
+  'ts-node-dev',
```
Comment (samchungy) [inline src/cli/configure/dependencies/skubaDeps.ts:8]: do we remove this now?

Comment (samchungy) [reply src/cli/configure/dependencies/skubaDeps.ts:8]: Oh nvm

**Review by samchungy — APPROVED** [review]
Looks aight from a quick look

File: src/cli/node.ts [hunk 2]
```diff
@@ -41,13 +36,6 @@ export const node = async () => {
     );
   }
 
-  // REPL with `ts-node` to support import statements.
-  return tsNode
-    .createRepl({
-      service: tsNode.register({
-        require: ['dotenv/config', 'tsconfig-paths/register'],
```
Comment (72636c) [inline src/cli/node.ts:48]: Is the thinking that these don't matter for a REPL, or is the functionality baked in to tsx? (if we're keeping a REPL around, being able to `import 'src/api/blah'` could still be nice...)

**Review by 72636c — APPROVED** [review]

File: src/cli/node.ts (hunk 2 above)
Comment (AaronMoat) [reply src/cli/node.ts:48]: There is no thinking 😂 I need to test this still. Good question though

File: .changeset/empty-cherries-admire.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+'skuba': major
+---
+
```
Comment (AaronMoat) [inline .changeset/empty-cherries-admire.md:4]: @72636c would you mind wordsmithing this? I'll proceed to merge without significant manual testing in SEEK repos; I hope to be able to get out https://github.com/seek-oss/skuba/pull/1537 as a second major changeset and bundle them in a release, so testing in real repos is likely to be valuable as a combined exercise

**Pull request merged**